docs/src/literate-tutorials/dg_heat_equation.jl

```diff
@@ -0,0 +1,334 @@
+    ## Reset to 0
+    fill!(fe, 0)
+    ## Loop over quadrature points
+    for q_point in 1:getnquadpoints(fv)
+        ## Get the normal to face A
+        normal = getnormal(fv, q_point)
+        ## Get the quadrature weight
+        ∂Ω = getdetJdV(fv, q_point)
+        ## Loop over test shape functions
+        for i in 1:getnbasefunctions(fv)
+            δu = shape_value(fv, q_point, i)
+            boundary_flux = normal[2]
+            fe[i] = boundary_flux * δu * ∂Ω
+        end
+    end
+    return fe
+end
+#md nothing # hide
+
+# #### Global assembly
+# We define the function `assemble_global` to loop over all elements and internal faces (interfaces), as well as the external faces involved in Neumann boundary conditions. 
+
+function assemble_global(cellvalues::CellValues, facevalues::FaceValues, interfacevalues::InterfaceValues, K::SparseMatrixCSC, dh::DofHandler, h::Float64)
+    ## Allocate the element stiffness matrix and element force vector
+    n_basefuncs = getnbasefunctions(cellvalues)
+    Ke = zeros(n_basefuncs, n_basefuncs)
+    fe = zeros(n_basefuncs)
+    Ki = zeros(n_basefuncs * 2, n_basefuncs * 2)
+    ## Allocate global force vector f
+    f = zeros(ndofs(dh))
+    ## Create an assembler
+    assembler = start_assemble(K, f)
+    ## Loop over all cells
+    for cell in CellIterator(dh)
+        ## Reinitialize cellvalues for this cell
+        reinit!(cellvalues, cell)
+        ## Compute volume integral contribution
+        assemble_element!(Ke, fe, cellvalues)
+        ## Assemble Ke and fe into K and f
+        assemble!(assembler, celldofs(cell), Ke, fe)
+    end
+    ## get interpolation dimensions and order to use in the penalty weight, using the $(1 + order)^{dim}$ for $\eta$
+    ip = Ferrite.function_interpolation(facevalues)
+    order = Ferrite.getorder(ip)
+    dim = Ferrite.getdim(ip)
+    μ = (1 + order)^dim / h
+    ## Loop over all interfaces
+    for ic in InterfaceIterator(dh)
+        ## Reinitialize interfacevalues for this interface
+        reinit!(interfacevalues, ic)
+        ## Compute interface surface integrals contribution
+        assemble_interface!(Ki, interfacevalues, μ)
+        ## Assemble Ki into K
+        assemble!(assembler, interfacedofs(ic), Ki)
+    end
+    ## Loop over domain boundaries with Neumann boundary conditions
+    for fc in FaceIterator(dh, ∂Ωₙ)
+        ## Reinitialize face_values_a for this boundary face
+        reinit!(facevalues, fc)
+        ## Compute boundary face surface integrals contribution
+        assemble_boundary!(fe, facevalues)
+        ## Assemble fe into f
+        assemble!(f, celldofs(fc), fe)
+    end
+    return K, f
+end
+#md nothing # hide
+
+# ### Solution of the system
+# The solution of the system is independent of the discontinuous discretization, and after assembling the application of constraints, linear solve, and exporting is done as usual.
+
+K, f = assemble_global(cellvalues, facevalues, interfacevalues, K, dh, h);
```

Comment:
Sorry, my writing from last time was a bit unclear. What do you think about moving it around like this?
```suggestion
K, f = assemble_global(cellvalues, facevalues, interfacevalues, K, dh, h);
#md nothing # hide

# ### Solution of the system
# The solution of the system is independent of the discontinuous discretization and the application of constraints, linear solve, and exporting is done as usual.

```